apps/openssl.c

```diff
@@ -130,22 +131,18 @@ static size_t internal_trace_cb(const char *buf, size_t cnt,
 {
     int ret = 0;
     tracedata *trace_data = vdata;
-    union {
-        CRYPTO_THREAD_ID tid;
-        unsigned long ltid;
-    } tid;
-    char buffer[256];
+    char buffer[256 + sizeof(CRYPTO_THREAD_ID)];
+    size_t n;
 
     switch (cmd) {
     case OSSL_TRACE_CTRL_BEGIN:
         if (!ossl_assert(!trace_data->ingroup))
             return 0;
         trace_data->ingroup = 1;
 
-        tid.ltid = 0;
-        tid.tid = CRYPTO_THREAD_get_current_id();
-
-        BIO_snprintf(buffer, sizeof(buffer), "TRACE[%lx]:%s: ", tid.ltid,
+        n = thread_to_string(buffer, sizeof(buffer),
+                             CRYPTO_THREAD_get_current_id(), "TRACE[", "]:");
+        BIO_snprintf(buffer + n, sizeof(buffer) - n, "%s: ",
                      OSSL_trace_get_category_name(category));
```

Comment:
I would like to suggest something like this:

``` C
    CRYPTO_THREAD_ID tid = CRYPTO_THREAD_get_current_id();
    char *hex = OPENSSL_buf2hexstr(&tid, sizeof(tid));
    BIO_snprintf(buffer, sizeof(buffer), "TRACE[%s]:%s: ",
                 hex, OSSL_trace_get_category_name(category));
    OPENSSL_free(hex);
```

(variable declarations will, of course, happen *elsewhere*, but I hope I got my point through.

Reply:
Note that this also makes it immediately obvious to anyone reading the code what's getting printed.

Reply:
Good suggestion.
I'll update and remove the header.

Reply:
Fixed.